### PR TITLE
[SYCL] Implement SYCL/device-requirements (pt.1)

### PR DESCRIPTION
**Pull request by KornevNikita**
As part of optional kernel features implementation we need to create the "SYCL/device-requirements" property set and populate it with the "aspects" property. Since we haven't implemented device code split per-aspect part yet, this PR only covers "aspects" part of "SYCL/device-requirements" property set.
Co-authored-by: @AlexeySachkov

### CODE REVIEW

File: llvm/test/tools/sycl-post-link/device-requirements/device-requirements.ll [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_EUlvE_ = comdat any
+
+@__spirv_BuiltInWorkgroupId = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+@__spirv_BuiltInGlobalLinearId = external dso_local local_unnamed_addr addrspace(1) constant i64, align 8
+@__spirv_BuiltInWorkgroupSize = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+
+; Function Attrs: convergent mustprogress noinline norecurse optnone
+define weak_odr dso_local spir_kernel void @_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_EUlvE_() #0 comdat !kernel_arg_buffer_location !6 {
+entry:
+  call void @__itt_offload_wi_start_wrapper()
```
Comment (AlexeySachkov) [inline llvm/test/tools/sycl-post-link/device-requirements/device-requirements.ll:35]: Could you please re-generate this test using `-fno-sycl-instrument-device-code` flag? this should reduce the amount of LLVM IR significantly, making the test simpler

File: llvm/test/tools/sycl-post-link/device-requirements/device-requirements.ll [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+;   foo();
+; }
+; int main() {
+;   sycl::queue q;
+;   q.submit([&](sycl::handler &cgh) {
+;     cgh.single_task([=]() { bar(); });
+;   });
+; }
+
+; RUN: sycl-post-link -split=auto -o %s -o %t.files.table
```
Comment (AlexeySachkov) [inline llvm/test/tools/sycl-post-link/device-requirements/device-requirements.ll:14]: We should have a variation of this test with per-kernel split to ensure that we generate properties correctly for each resulting module

File: sycl/source/detail/device_binary_image.hpp [hunk 3]
```diff
@@ -207,6 +210,7 @@ class RTDeviceBinaryImage {
   RTDeviceBinaryImage::PropertyRange ProgramMetadata;
   RTDeviceBinaryImage::PropertyRange ExportedSymbols;
   RTDeviceBinaryImage::PropertyRange DeviceGlobals;
+  RTDeviceBinaryImage::PropertyRange DeviceRequirements;
```
Comment (AlexeySachkov) [inline sycl/source/detail/device_binary_image.hpp:213]: This is likely an ABI break. I suggest that we leave RT part (`sycl/`) subdirectory for a separate PR for now to minimize amount of changes which needs to be reviewed

File: llvm/test/tools/sycl-post-link/device-requirements/device-requirements.ll (hunk 1 above)
Comment (KornevNikita) [reply llvm/test/tools/sycl-post-link/device-requirements/device-requirements.ll]: Wow, good prompt, thanks.

File: sycl/source/detail/device_binary_image.hpp (hunk 3 above)
Comment (KornevNikita) [reply sycl/source/detail/device_binary_image.hpp]: Should we alse delete pi.h part for now?

Comment (AlexeySachkov) [reply sycl/source/detail/device_binary_image.hpp]: Yes, I suggest that we only leave `llvm/` part for now

**Pull request merged**